### PR TITLE
[SYCL] Fixed assert on windows with GetModuleFileNameA failed

### PR DESCRIPTION
**Pull request by alexanderfle**
This patch fixes a problem on Windows, when any unit test with invocation kernel run:
`Assertion failed: Ret > 0 && "GetModuleFileNameA failed", file D:\sycl_workspace\llvm\sycl\source\detail\os_util.cpp, line 208`

GetModuleFileNameA() requires 0 as the first parameter for current executable module, but it is -1 (OSUtil::ExeModuleHandle), see return value in getOSModuleHandle().

About GetModuleFileNameA() from Windows API doc :
> A handle to the loaded module whose path is being requested. If this parameter is NULL, GetModuleFileName retrieves the path of the executable file of the current process.

Signed-off-by: Alexander Flegontov <alexander.flegontov@intel.com>

### CODE REVIEW

File: sycl/source/detail/os_util.cpp [hunk 1]
```diff
@@ -191,7 +191,7 @@ OSModuleHandle OSUtil::getOSModuleHandle(const void *VirtAddr) {
     return 0;
   }
   if (PhModule == GetModuleHandleA(nullptr))
-    return OSUtil::ExeModuleHandle;
+    return 0;
```
Comment (kbobrovs) [inline sycl/source/detail/os_util.cpp:194]: Now caller won't know that it is the exe module handle. You probably need OSUtil::ExeModuleHandle to os-specific module handle conversion util function for 'exe' module and use it on the caller side, where os-specific API is used (AFAIU).

File: sycl/include/CL/sycl/detail/os_util.hpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ class __SYCL_EXPORT OSUtil {
 
   /// Module handle for the executable module - it is assumed there is always
   /// single one at most.
-  static constexpr OSModuleHandle ExeModuleHandle = -1;
+  static constexpr OSModuleHandle ExeModuleHandle = 0;
```
[the remark this answers is not in the document]
Comment (alexanderfle) [reply sycl/include/CL/sycl/detail/os_util.hpp]: ok

File: sycl/source/detail/os_util.cpp (hunk 1 above)
Comment (alexanderfle) [reply sycl/source/detail/os_util.cpp]: Thanks, I fixed

**Comment by alexanderfle** [comment]
@kbobrovs, I addressed your comments, could you please take a look again, and approve if it is ok?

**Pull request merged**